openassessment/xblock/test/test_staff.py

```diff
@@ -0,0 +1,126 @@
+
+        # Submit a staff-assessment
+        self.set_staff_access(xblock)
+        self.ASSESSMENT['submission_uuid'] = submission['uuid']
+        resp = self.request(xblock, 'staff_assess', json.dumps(self.ASSESSMENT), response_format='json')
+        self.assertTrue(resp['success'])
+
+        # Expect that a staff-assessment was created
+        assessment = staff_api.get_latest_assessment(submission['uuid'])
+        self.assertEqual(assessment['submission_uuid'], submission['uuid'])
+        self.assertEqual(assessment['points_earned'], 5)
+        self.assertEqual(assessment['points_possible'], 6)
+        self.assertEqual(assessment['scorer_id'], 'Bob')
+        self.assertEqual(assessment['score_type'], 'ST')
+        self.assertEqual(assessment['feedback'], u'')
+
+        parts = sorted(assessment['parts'])
+        self.assertEqual(len(parts), 2)
+        self.assertEqual(parts[0]['option']['criterion']['name'], u'Form')
+        self.assertEqual(parts[0]['option']['name'], 'Fair')
+        self.assertEqual(parts[1]['option']['criterion']['name'], u'𝓒𝓸𝓷𝓬𝓲𝓼𝓮')
+        self.assertEqual(parts[1]['option']['name'], u'ﻉซƈﻉɭɭﻉกՇ')
+
+        # get the assessment scores by criteria
+        assessment_by_crit = staff_api.get_assessment_scores_by_criteria(submission["uuid"])
+        self.assertEqual(assessment_by_crit[u'𝓒𝓸𝓷𝓬𝓲𝓼𝓮'], 3)
+        self.assertEqual(assessment_by_crit[u'Form'], 2)
+
+        score = staff_api.get_score(submission["uuid"], None)
+        self.assertEqual(assessment['points_earned'], score['points_earned'])
+        self.assertEqual(assessment['points_possible'], score['points_possible'])
+
+    @scenario('data/self_assessment_scenario.xml', user_id='Bob')
+    def test_staff_assess_permission_error(self, xblock):
+        # Create a submission for the student
+        student_item = xblock.get_student_item_dict()
+        xblock.create_submission(student_item, self.SUBMISSION)
+        resp = self.request(xblock, 'staff_assess', json.dumps(self.ASSESSMENT))
+        self.assertIn("You do not have permission", resp)
+
+    @scenario('data/self_assessment_scenario.xml', user_id='Bob')
+    def test_staff_assess_invalid_options(self, xblock):
+        student_item = xblock.get_student_item_dict()
+
+        # Create a submission for the student
+        submission = xblock.create_submission(student_item, self.SUBMISSION)
+
+        self.set_staff_access(xblock)
+        self.ASSESSMENT['submission_uuid'] = submission['uuid']
+
+        for key in self.ASSESSMENT:
+            assessment_copy = copy.copy(self.ASSESSMENT)
+            del assessment_copy[key]
+            resp = self.request(xblock, 'staff_assess', json.dumps(assessment_copy), response_format='json')
+            self.assertFalse(resp['success'])
+            self.assertIn('msg', resp)
+
+    @scenario('data/self_assessment_scenario.xml', user_id='bob')
+    def test_staff_assess_assessment_error(self, xblock):
```

Comment:
Nit: ``test_staff_assess_assessment_error`` is hard to read. Given that all these tests are inside the ``TestStaffAssessment`` class, I think they could lose the redundant ``staff_access`` portion of their names.